### DISCUSSION TITLE
关于milvus集群的一些疑惑

### DISCUSSION THREAD

**Discussion by jilieryuyi** [milvus-io/milvus#4950] [Q&A and General discussion]
为什么采用共享存储的方式实现？而不是将数据hash到不同的节点？ 或者其他分片方案！将数据打散到不同的节点？这么做的出发点是什么，有什么好处？

**Comment by yhmo** [comment]
Mishards是一个分片中间件，在设计之初主要考虑不改变milvus原有结构和使用方式，基本上以黑盒的方式来包装milvus。由1个write节点负责数据的写入和管理，mishards只做请求转发和任务分发。采用共享存储的方式，让所有节点都能读取同一份数据，而不是把数据分散存储在多个节点上（那样做会把mishards的逻辑弄复杂）。mishards会将数据hash到不同的节点，那是把多个分片hash到多个节点的内存中以供搜索。
概括来说就是，共享存储让数据持久化在同一个地方，搜索时，再把数据分片hash到多个节点的内存中。

**Comment by yhmo** [comment]
Mishards仅仅是一个分片中间件，而milvus真正的分布式版本(v2.0)会在近期开源。

**Reply by jilieryuyi** [reply]
> Mishards仅仅是一个分片中间件，而milvus真正的分布式版本(v2.0)会在近期开源。

期待2.0
